src/Microsoft.DotNet.Tools.Compiler/CompilerCommandApp.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+using System.Linq;
+
+//This class is responsible with defining the arguments for the Compile verb.
+//It knows how to interpret them and set default values
+namespace Microsoft.DotNet.Tools.Compiler
+{
+    public delegate bool OnExecute(List<ProjectContext> contexts, CompilerCommandApp compilerCommand);
+
+    public class CompilerCommandApp
+    {
+        private readonly CommandLineApplication _app;
+
+        //options and arguments for compilation
```

Comment:
I know this is copied from the old Compile logic... but it's pretty gross. We need to come up with a better solution [outside of this PR]